docs/architecture/mxgraph-integration.adoc

```diff
@@ -0,0 +1,89 @@
+=== mxGraph Integration
+
+The https://jgraph.github.io/mxgraph/[mxGraph] integration is in charge of displaying the BPMN diagrams, by filling the
+https://jgraph.github.io/mxgraph/docs/manual.html#3.1.1:[mxGraph model] from the BPMN model.
+
+It defines the https://jgraph.github.io/mxgraph/docs/manual.html#3.1.3.1[BPMN styles] including dedicated BPMN shapes
+which are associated to the `mxGraph` https://jgraph.github.io/mxgraph/docs/manual.html#3.1.3.4[mxCell] when the elements
+are inserted into the `mxGraph` model:
+
+* `vertex` for BPMN Shapes
+* `edge` for BPMN Edges
+
+The main entry point for `mxGraph` rendering is `MxGraphRenderer`.
+
+For more technical details, see <<bpmn-support-howto-elements-rendering,BPMN Elements rendering>>.
+
+If you are new to `mxGraph` or want to know more about it, please have a look at the https://jgraph.github.io/mxgraph/docs/tutorial.html[mxGraph tutorial]
+and https://jgraph.github.io/mxgraph/docs/manual.html[mxGraph manual]
+
+
+==== Logical Groups of BPMN Elements
+
+https://jgraph.github.io/mxgraph/docs/manual.html#3.1.4[mxGraph Group Structure] allows to define a parent-child relationship
+that leverages various rendering features like collapsing/expanding, layering, drill down, group move/hide, ...
+See https://jgraph.github.io/mxgraph/docs/manual.html#3.1.5[mxGraph Complexity Management] for more details.
+
+On `mxGraph` graph creation, the lib fills the following group of BPMN elements:
+
+* `pool` is the parent of `lanes` and `inter-lane elements` (for instance, `message flows`)
+* `lane` is the parent of elements it includes in the BPMN source
+
+The default mxGraph parent is the parent of
+
+* inter pool elements (for instance, `message flows`)
+* elements when the BPMN source doesn't define any lane/pool
+
+
+==== BPMN elements positioning
+
+*Note*: the following applies to both BPMN shapes and edges (including waypoints)
+
+The bpmn input source coordinates are absolute whereas mxgraph uses coordinates in the referential of the parent cell.
+As we define a parent-child relationship when inserting BPMN elements in the mxGraph model, a coordinate transformation
+layer is required. See `MxGraphRenderer` for more details.
```

Comment:
```suggestion
layer is required. See https://github.com/process-analytics/bpmn-visualization-js/blob/d812e36bca8dea8f9275161f2eb9a9a5173a9ca5/src/component/mxgraph/MxGraphRenderer.ts[MxGraphRenderer] for more details.
```

Reply:
We previously decided to not link the doc to the source code. So I did the same as in the 'bpmn support how to' section: just refer to the class/file name